### PR TITLE
GitHub Authentication, daily page and export button

### PR DESCRIPTION
**Pull request by sdelamo**
- change package from io.micronaut.starter.analytics.postgres to io.micronaut.starter.analytics
- Adds OpenAPI specification generation and exposure
- Authentication with GitHub 
- Increase page size for excel generation
- Daily and Percentages with last 30 days. 
- Organize code in well-defined packages: 

- `configuration`
- `controllers`
- `entities`
- `gcp`
-`repositories`
-`dev`
-`services`
-`security`

### CODE REVIEW

File: src/main/resources/views/percentages.html [hunk 1]
```diff
@@ -23,6 +23,9 @@ <h1 class="title title_large first-word-bold first-word-break"><strong th:text="
                 <div th:replace="~{_piecharts :: piecharts(${charts.subList(1,charts.size())})}"></div>
             </div>
         </div>
+        <th:block th:if="${security}">
+            <div th:replace="~{_excel :: excel()}"></div>
+        </th:block>
```
Comment (timyates) [inline src/main/resources/views/percentages.html]: If we move this up into the `col-md-4.grid-item.inner` above (with the title of the page) then it looks like

image

Instead...

Which I kinda prefer, but YMMV 😉

**Review by timyates — APPROVED** [review]

File: src/main/java/io/micronaut/starter/analytics/postgres/security/ApiKeyTokenReader.java [hunk 2]
```diff
@@ -46,6 +48,14 @@ protected String getHeaderName() {
 
     @Override
     public Optional<String> findToken(HttpRequest<?> request) {
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Looking for bearer token in parameter {}", PARAMETER_API_KEY);
+        }
+        Optional<String> tokenOptional = Optional.ofNullable(request.getParameters().get(PARAMETER_API_KEY));
+        if (tokenOptional.isPresent()) {
+            return tokenOptional;
+        }
```
Comment (timyates) [inline src/main/java/io/micronaut/starter/analytics/postgres/security/ApiKeyTokenReader.java]: This has also been done in the daily PR, but in there the query param is checked AFTER the header

https://github.com/micronaut-projects/micronaut-starter-analytics/pull/15/files#diff-77beeaee4a39ac2dd16ad4b71395ee5254318f81eb151ead9cbeec47feda1745

Not sure which is better 🤔

Comment (sdelamo) [reply src/main/java/io/micronaut/starter/analytics/postgres/security/ApiKeyTokenReader.java]: I have removed the queryValue parameter and added GitHub Authentication in this PR.

**Review by timyates — APPROVED** [review]
Cool 👍 Functionally tested and updated the instructions

**Pull request merged**